apps/analog-app/src/app/pages/(home).server.ts

```diff
@@ -0,0 +1,12 @@
+import { PageServerLoad } from '@analogjs/router';
+import { Product } from '../products';
+
+export const load = async ({ fetch }: PageServerLoad) => {
+  const products = await fetch<Product[]>('/api/v1/products');
+
+  return {
+    products: products,
+  };
+};
+
+export type Load = Awaited<ReturnType<typeof load>>;
```

Comment:
How do you feel about exposing a type helper like below:
```ts
export type LoadReturn<
  A extends (pageServerLoad: PageServerLoad) => Promise<any>
> = Awaited<ReturnType<A>>;
```
So one can simply write: `export type Load = LoadReturn<typeof load>`. This might help people who don't know about Awaited and ReturnType

Reply:
Nice. Can we make this more direct and not have to declare the type? My advanced TypeScript is not great 😅 

```ts
import { Component } from '@angular/core';
import { toSignal } from '@angular/core/rxjs-interop';
import { RouteMeta, useLoad } from '@analogjs/router';
import { NgForOf, NgIf } from '@angular/common';
import { RouterLinkWithHref } from '@angular/router';

import { ProductAlertsComponent } from '../product-alerts/product-alerts.component';

import type { load } from './(home).server';

export const routeMeta: RouteMeta = {
  title: 'Product List',
};

@Component({
  selector: 'app-product-list',
  standalone: true,
  imports: [NgForOf, NgIf, ProductAlertsComponent, RouterLinkWithHref],
  template: `
    <h2>Products</h2>

    <div *ngFor="let product of data().products">
      <h3>
        <a
          [title]="product.name + ' details'"
          [routerLink]="['/products', product.id]"
        >
          {{ product.name }}
        </a>
      </h3>

      <p *ngIf="product.description">Description: {{ product.description }}</p>

      <button type="button" (click)="share()">Share</button>

      <app-product-alerts [product]="product" (notify)="onNotify()">
      </app-product-alerts>
    </div>
  `,
  styles: [
    `
      $neon: lightblue;

      @mixin background($color: #fff) {
        background: $color;
      }

      h2 {
        @include background($neon);
      }
    `,
  ],
})
export default class ProductListComponent {
  data = toSignal(useLoad<typeof load>(), { requireSync: true });

  share() {
    window.alert('The product has been shared!');
  }

  onNotify() {
    window.alert('You will be notified when the product goes on sale');
  }
}

```

Reply:
This should work:

```ts
import { Injector, assertInInjectionContext, inject } from '@angular/core';
import { ActivatedRoute, Data } from '@angular/router';
import { map } from 'rxjs';
import { PageServerLoad } from './route-types';

export function useLoad<
  T extends (pageServerLoad: PageServerLoad) => Promise<any>
>(options?: { injector?: Injector }) {
  !options?.injector && assertInInjectionContext(useLoad);

  const injector = options?.injector ?? inject(Injector);
  const route = injector.get(ActivatedRoute);

  return route.data.pipe(
    map<Data, Awaited<ReturnType<T>>>((data) => data['load'])
  );
}
```

Reply:
🔥